src/packet/tcp.rs

```diff
@@ -0,0 +1,197 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! TCP packet abstraction
+
+use packet::Packet;
+use packet::HasPseudoheader;
+use packet::checksum::rfc1071_checksum;
+use pnet_macros::types::*;
+
+/// Represents a TCP Packet
+#[packet]
+pub struct Tcp {
```

Comment:
I haven't checked the structure of this packet to make sure it matches the spec - I'll do that in a later review.